src/system-health/health_checker/sysmonitor.py

```diff
@@ -0,0 +1,634 @@
+                            #Using shell=True is unsafe in general and should be avoided
+                            shell=(not safe),
+                            universal_newlines=True,
+                            stderr=(subprocess.STDOUT if not keep_std_err else subprocess.PIPE))
+    try:
+        stdout, error = proc.communicate(timeout=30)
+        proc.wait()
+        return stdout, error
+
+    except subprocess.TimeoutExpired:
+        logger.log_error("TimeoutExpired for {}".format(command))
+
+
+class MonitorStateDbTask(object):
+    def __init__(self):
+        self.task_queue = mpmgr.Queue()
+        self.task_process = None
+        self.task_stopping_event = multiprocessing.Event()
+
+    def __init__(self,myQ):
+        self.task_queue = myQ
+        self.task_process = None
+        self.task_stopping_event = multiprocessing.Event()
+
+    def task_worker_statedb(self):
+        #logger.log_info("Start monitoring statedb feature table (pid {0})".format(os.getpid()))
+        update_proc_name('sysmonitor_statedb')
+        from swsscommon import swsscommon
+        state_db = swsscommon.DBConnector("STATE_DB", REDIS_TIMEOUT_MS, True)
+        sel = swsscommon.Select()
+        cst = swsscommon.SubscriberStateTable(state_db, "FEATURE")
+        sel.addSelectable(cst)
+            
+        while not self.task_stopping_event.is_set():
+            (state, c) = sel.select(SELECT_TIMEOUT_MSECS)
+            if state == swsscommon.Select.TIMEOUT:
+                continue
+            if state != swsscommon.Select.OBJECT:
+                logger.log_warning("sel.select() did not return swsscommon.Select.OBJECT")
+                continue
+            (key, op, cfvs) = cst.pop()
+            key_ext = key+".service"
+            #logger.log_info("featureevent received:{}".format(key_ext))
+            timestamp="{}".format(datetime.datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S"))
+            msg={"unit": key_ext,"evt_src":"feature","time":timestamp}
+            self.task_notify(msg)
+            time.sleep(0.1)
+        
+        #logger.log_info("Stop STATE_DB monitoring loop")
+        
+    def task_worker(self):
+        while not self.task_stopping_event.is_set():
+            try:
+                self.task_worker_statedb()
+            except Exception as e:
+                logger.log_error("task_worker_statdb exited- {}, retry".format(str(e)))
+            time.sleep(1)
+    
+
+    def task_run(self):
+        if self.task_stopping_event.is_set():
+            return
+        self.task_process = multiprocessing.Process(target=self.task_worker)
+        self.task_process.start()
+
+    def task_notify(self, msg):
+        if self.task_stopping_event.is_set():
+            return
+        self.task_queue.put(msg)
+        time.sleep(0.001)
+
+    def task_stop(self):
+        self.task_stopping_event.set()
+        self.task_process.join()
+
+
+class MonitorSystemBusTask(object):
```

Comment:
Could you please consider inheriting from ProcessTaskBase in src/sonic-py-common/sonic_py_common/task_base.py? BTW, what is the purpose to use a process instead of a thread?

Reply:
Done